runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -1017,21 +1294,229 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
+
+	_edenSizeFactor = (intptr_t)targetEdenRegions - currentIdealEdenRegions;
+}
+
+void
+MM_SchedulingDelegate::checkEdenSizeAfterPgc(MM_EnvironmentVLHGC *env, bool globalSweepHappened)
+{
+	if (!_extensions->statupPhaseFinished) {
+		/* Don't change eden size during startup phase - keep it at default */
+		return;
+	}
+
+	if (_currentlyPerformingGMP && !globalSweepHappened) {
+		/* 
+		 * Don't change eden size while GMP cycle is running - 
+		 * Unless a global sweep just happened, in which case we allow eden to change on first PGC after GMP 
+		 */
+		return;
+	}
+
+	if (heapIsFullyExpanded(env)) {
+		/* 
+		 * When heap is full, try to get eden to such a size to minimize overhead, while keeping into account the limits imposed by
+		 * the heap size, and how much free space is available, since "blindly" increasing eden is no longer an option
+		 */
+		if (globalSweepHappened) {
+			/* 
+			 * Take a more aggreessive step towards ideal eden. 
+			 * At this point we have the most accuate information about liveness in the heap, so we make the most informed decision
+			 */
+			moveTowardRecommendedEden(env, 0.5);
+			resetPgcTimeStatistics(env);
+		} else if ((_pgcCountSinceGMPEnd & (consecutivePGCToChangeEden - 1)) == 0) {
+			/* Every consecutivePGCToChangeEden number of pgc's, re-evaluate eden size, and move towards it */
+			moveTowardRecommendedEden(env, 0.25);
+		}
+	} else if (0 == _pgcCountSinceGMPEnd % 3) {
+		/*
+		 * On every third pgc, make an adjustment to eden size based on observed pgc values. 
+		 * Waiting for every third PGC, allows some statistics (which are being averaged) to settle to their new true averages
+ 		 */
+
+		intptr_t edenRegionChange = 0;
+		intptr_t edenChangeMagnitude = (intptr_t)ceil((0.1 * getIdealEdenSizeInBytes(env)) / _regionManager->getRegionSize());
+
+		double hybridEdenOverhead = calculateHybridEdenOverhead(env, _historicalPartialGCTime, _partialGcOverhead);
+		
+		/* 
+		 * Aim to get hybrid PGC overhead between extensions->dnssExpectedTimeRatioMinimum and extensions->dnssExpectedTimeRatioMaximum 
+		 * by increasing or decreasing eden by 10% 
+		 */
+		if (_extensions->dnssExpectedTimeRatioMinimum._valueSpecified > hybridEdenOverhead ) {
+			/* Shrink eden a bit */
+			edenRegionChange = edenChangeMagnitude * -1;
+		} else if (_extensions->dnssExpectedTimeRatioMaximum._valueSpecified < hybridEdenOverhead) {
+			/* Expand eden a bit */
+			edenRegionChange = edenChangeMagnitude;
+		}
+
+		_edenSizeFactor += edenRegionChange;	
+	}
+}
+
+double 
+MM_SchedulingDelegate::mapPgcTimeToPgcOverhead(MM_EnvironmentVLHGC *env, uintptr_t partialGcTimeMs) {
+	
+	/* Convert expectedTimeRatioMinimum/Maximum to 0-100 based for this formula */
+	double xminpct = _extensions->dnssExpectedTimeRatioMinimum._valueSpecified * 100;
+	double xmaxpct = _extensions->dnssExpectedTimeRatioMaximum._valueSpecified * 100;
+	double xmaxt = (double)_extensions->tarokTargetMaxPauseTime;
+
+	double overhead;
```

Comment:
As per coding standards, this should be initied to a neutral value. Then, use can set it to midpointPct, and simplify one of the case down below.